src/Api.php

```diff
@@ -0,0 +1,626 @@
+{
+    private Config $config;
+
+    private Client $client;
+
+    private string $accessToken;
+
+    public function __construct(Config $config, Client $client)
+    {
+        $this->config = $config;
+        $this->client = $client;
+    }
+
+    public function login(): string
+    {
+        try {
+            $response = $this->client->post(
+                sprintf('%s/api/v1/authentication/login', $this->config->getUrlAddress()),
+                [
+                    'form_params' => [
+                        'username' => $this->config->getUsername(),
+                        'password' => $this->config->getPassword(),
+                    ],
+                ]
+            );
+        } catch (ClientException $exception) {
+            throw new UserException($exception->getMessage(), $exception->getCode(), $exception);
+        } catch (ConnectException $exception) {
+            $exceptionMessage = $exception->getMessage();
+            if (strstr($exception->getMessage(), 'cURL error 6: Could not resolve host')) {
+                $exceptionMessage = sprintf('Could not resolve host "%s"', $this->config->getUrlAddress());
+            }
+            throw new UserException($exceptionMessage, $exception->getCode(), $exception);
+        }
+
+        $responseJson = json_decode($response->getBody()->getContents(), true);
+        $this->accessToken = $responseJson['access_token'];
+        return $this->accessToken;
+    }
+
+    public function uploadFile(string $dataDir, string $filename): string
+    {
+        try {
+            $streamFile = new MultipartStream(
+                [
+                    [
+                        'name' => 'file',
+                        'filename' => $filename,
+                        'contents' => file_get_contents($dataDir . $filename),
+                    ],
+                ]
+            );
+            $response = $this->client->request(
+                'post',
+                sprintf('%s/storage/fs/upload', $this->config->getUrlAddress()),
+                [
+                    'headers' => [
+                        'Authorization' => 'Bearer ' . $this->accessToken,
+                        'x-upload-token' => $this->validateFile($dataDir, $filename),
+                        'Content-Type' => 'multipart/form-data; boundary=' . $streamFile->getBoundary(),
+                    ],
+                    'body' => $streamFile,
+                ]
+            );
+        } catch (ClientException $exception) {
+            throw new UserException($exception->getMessage(), $exception->getCode(), $exception);
+        } catch (ServerException $exception) {
+            throw new UserException($exception->getMessage(), $exception->getCode(), $exception);
+        }
+
+        $responseJson = json_decode($response->getBody()->getContents(), true);
+        return $responseJson[0]['storageInfo']['path'];
+    }
+
+    public function checkDatamodelExists(string $datamodelName): ?array
```

Comment:
Nazov `checkDatamodelExists` mi evokuje, ze by to malo vratit `bool`, ... alebo potom upravit nazov na `getDatamodel` ?

Reply:
jj, všechny jsem je přejmenoval 👍 díky